parts/kubernetesmastercustomscript.sh

```diff
@@ -92,6 +93,43 @@ function ensureKubectl() {
     fi
 }
 
+function setNetworkPlugin () {
+    sed -i "s/^KUBELET_NETWORK_PLUGIN=.*/KUBELET_NETWORK_PLUGIN=${1}/" /etc/default/kubelet
+}
+
+function setDockerOpts () {
+    sed -i "s#^DOCKER_OPTS=.*#DOCKER_OPTS=${1}#" /etc/default/kubelet
+}
+
+function configNetworkPolicy() {
+    if [[ ! -z "${APISERVER_PRIVATE_KEY}" ]]; then
+        # on masters
+        ADDONS="calico-configmap.yaml calico-daemonset.yaml"
+        ADDONS_PATH=/etc/kubernetes/addons
+        CALICO_URL="https://raw.githubusercontent.com/projectcalico/calico/a4ebfbad55ab1b7f10fdf3b39585471f8012e898/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend-addon-manager"
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                curl -sSL --retry 12 --retry-delay 10 "${CALICO_URL}/${addon}" > "${ADDONS_PATH}/${addon}"
+            done
+        else
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                rm -f "${ADDONS_PATH}/${addon}"
+            done
+        fi
+    else
+        # on agents
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
```

Comment:
Sorry @colemickens I am UK based, so it was a bit too late for me

Basically the way this works even though the `kubelet`s don't have `kubenet` set, is that the routes for the individual node's Pod IP range are setup in Azure (and GCE as @lxpollitt points out) by the `controller-manager` ([configured here](https://github.com/Azure/acs-engine/blob/master/parts/kubernetesmaster-kube-controller-manager.yaml#L19)) as long as cloud-provider is set. No special config on the `kubelet` is needed (IMHO).  

This will continue to work and allocate from the cluster wide Pod network. Calico then just takes the IP addresses out every nodes LocalPodRange. (this [setting](https://github.com/projectcalico/calico/blob/a4ebfbad55ab1b7f10fdf3b39585471f8012e898/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend-addon-manager/calico-configmap.yaml#L20))

If you want to have a chat on Monday, we can set something up in your morning.